### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by nolanlawson**
Standard dependency update.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,24 +37,25 @@
...
-    "@commitlint/cli": "^14.1.0",
-    "@commitlint/config-conventional": "^14.1.0",
-    "@rollup/plugin-replace": "^3.0.0",
-    "@rollup/plugin-typescript": "^8.2.5",
-    "@types/jest": "^27.0.1",
-    "@types/node": "^16.7.13",
-    "execa": "^5.1.1",
-    "husky": "^7.0.4",
+    "@commitlint/cli": "^17.0.2",
+    "@commitlint/config-conventional": "^17.0.2",
```
Comment (nolanlawson) [inline package.json:41]: Checked the release notes, nothing in commitlint's changes should affect us.

File: package.json [hunk 2]
```diff
@@ -37,24 +37,25 @@
...
+    "@rollup/plugin-replace": "^4.0.0",
+    "@rollup/plugin-typescript": "^8.3.3",
+    "@types/jest": "^28.1.2",
+    "@types/node": "^18.0.0",
+    "execa": "^6.1.0",
```
Comment (nolanlawson) [inline package.json:46]: We're ESM now, so execa v6 is fine.

File: package.json [hunk 3]
```diff
@@ -37,24 +37,25 @@
...
     "is-ci": "^3.0.0",
-    "jest": "^27.1.0",
-    "lint-staged": "^11.2.6",
-    "prettier": "^2.4.1",
-    "rollup": "^2.56.3",
-    "semver": "^7.3.5",
-    "ts-jest": "^27.0.5",
-    "tslib": "^2.3.1",
+    "jest": "^28.1.1",
+    "jest-environment-jsdom": "^28.1.1",
```
Comment (nolanlawson) [inline package.json:50]: Needed in Jest v28.

File: package.json [hunk 4]
```diff
@@ -37,24 +37,25 @@
...
-    "@commitlint/config-conventional": "^14.1.0",
-    "@rollup/plugin-replace": "^3.0.0",
-    "@rollup/plugin-typescript": "^8.2.5",
-    "@types/jest": "^27.0.1",
-    "@types/node": "^16.7.13",
-    "execa": "^5.1.1",
-    "husky": "^7.0.4",
+    "@commitlint/cli": "^17.0.2",
+    "@commitlint/config-conventional": "^17.0.2",
+    "@rollup/plugin-replace": "^4.0.0",
```
Comment (nolanlawson) [inline package.json:42]: Seems unlikely this would break us: https://github.com/rollup/plugins/blob/master/packages/replace/CHANGELOG.md#v400

**Review by ravijayaramappa — APPROVED** [review]

**Pull request merged**